stockApp/stockApp/View/CameraViewController.swift

```diff
@@ -297,3 +323,45 @@ extension CameraViewController: AVCaptureVideoDataOutputSampleBufferDelegate {
 		}
 	}
 }
+
+// MARK: - UIPickerViewDelegate/UIPickerViewDataSource
+extension CameraViewController: UIPickerViewDelegate, UIPickerViewDataSource {
+
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return Ints.numberOfComponents
+    }
+
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return filters.count
+    }
+
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        return filters[row]
+    }
+
+    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        selectedFilter = filters[row]
+    }
+}
+
+// MARK: - SearchAlertViewDelegate
+extension CameraViewController: SearchAlertViewDelegate {
+
+    func searchButtonPressed(from view: SearchAlertView) {
+        self.searchAlertView.processSearch()
+    }
+
+    func cancelButtonPressed() {
+        self.searchAlertView.dismissAlert()
```

Comment:
```suggestion
        searchAlertView.dismissAlert()
```
only needed if you want to give context